WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -234,6 +235,12 @@ private ProcessedResult ProcessNoLock(SmartTransaction tx)
 				// If we did not have it.
 				if (Coins.TryAdd(newCoin))
 				{
+					// If the coin was banned from coinjoining before.
+					if (Coins.IsCoinBanned(newCoin, DateTimeOffset.Now, out DateTimeOffset? bannedUntil))
+					{
+						newCoin.BannedUntilUtc = bannedUntil;
+					}
```

Comment:
This is a responsability of the `CoinRegistry` and NOT the `TransactionProcessor` . The idea of a good design is that every component has a responsability and that makes easy to limit the number of changes, otherwise all features would need to touch all components. 

`TransactionProcessor` logic has nothing to do banning coins and then it shouldn't need to change at all. Please rever and add this code where it belongs to: `Coins::TryAdd`